src/cockpit/transports.py

```diff
@@ -143,7 +144,12 @@ def _write_eof_now(self) -> None:
 
     def _write_ready(self):
         assert self._queue is not None
-        n_bytes = os.writev(self._queue)
+
+        try:
+            n_bytes = os.writev(self._queue)
+        except BrokenPipeError:
+            self.abort()
+            return
```

Comment:
On line 166 we have:

```
            if self._closing:
                self.abort()
```

As `self.abort` now sets `self._closing` but I guess this needs to be conditional.

Reply:
Note, that `self._consider_closing()` seems to not be implemented so that sounds like a bug!

Reply:
It is a bug!  I think we don't fail the tests because we never test a situation where the write queue backs up enough to trigger this issue.

On the other hand: this might be exactly why a few random testcases fail for the injector.  I'll try to figure out how to test this...